src/scripts/main.js

```diff
@@ -1,3 +1,274 @@
+    if (newRow[i] === newRow[i + 1]) {
+      const newValue = newRow[i] * 2;
+
+      newRow[i] = newValue;
+      newRow[i + 1] = 0;
+
+      score += newRow[i];
+      checkResult(newValue);
+    }
+  }
+
+  const rowResult = filterZeros(newRow);
+
+  while (rowResult.length < COLUMNS) {
+    rowResult.push(0);
+  }
+
+  return rowResult;
+};
+
+const slideLeft = (currentBoard) => {
+  for (let i = 0; i < ROWS; i++) {
+    let row = currentBoard[i];
+
+    row = slide(row);
+    currentBoard[i] = row;
+  }
+};
+
+const slideRight = (currentBoard) => {
+  for (let i = 0; i < ROWS; i++) {
+    let row = currentBoard[i];
+
+    row = slide(row.reverse());
+    currentBoard[i] = row.reverse();
+  }
+};
+
+const slideUp = (currentBoard) => {
+  for (let i = 0; i < ROWS; i++) {
+    const verticalRow = [];
+
+    for (const arr of currentBoard) {
+      verticalRow.push(arr[i]);
+    }
+
+    const row = slide(verticalRow);
+
+    for (let k = 0; k < currentBoard.length; k++) {
+      currentBoard[k][i] = row[k];
+    }
+  }
+};
+
+const slideDown = (currentBoard) => {
+  for (let i = 0; i < ROWS; i++) {
+    const verticalRow = [];
+
+    for (const arr of currentBoard) {
+      verticalRow.push(arr[i]);
+    }
+
+    const row = slide(verticalRow.reverse());
+
+    row.reverse();
+
+    for (let k = 0; k < currentBoard.length; k++) {
+      currentBoard[k][i] = row[k];
+    }
+  }
+};
+
+const getNewCell = () => {
+  const percent = Math.floor(Math.random() * 100) + 1;
+
+  switch (true) {
+    case percent <= 10:
+      return 4;
+
+    default:
+      return 2;
+  }
+};
+
+const addNewCell = () => {
+  const emptyCells = getEmptyCells();
+  const newCellValue = getNewCell();
+
+  const randomEmptyCell = _.sample(emptyCells);
+
+  if (gameBoard[randomEmptyCell.r][randomEmptyCell.c] === 0) {
+    gameBoard[randomEmptyCell.r][randomEmptyCell.c] = newCellValue;
+  }
+};
+
+const prepareStart = () => {
+  messageStart.style.display = 'none';
+  startButton.innerText = 'Restart';
+  startButton.style.fontSize = '18px';
+  startButton.classList.remove('start');
+  startButton.classList.add('restart');
+};
+
+const restartGame = () => {
+  gameBoard = JSON.parse(JSON.stringify(initialBoard));
+  messageWin.classList.add('hidden');
+  messageLose.classList.add('hidden');
+  score = 0;
+  isWon = false;
+  document.removeEventListener('keyup', gameControl);
+  document.addEventListener('keyup', gameControl);
+};
+
+const checkMerges = () => {
+  const gameBoardPrevState = JSON.parse(JSON.stringify(gameBoard));
+  const gameBoardWillModified = JSON.parse(JSON.stringify(gameBoard));
+
+  const actions = [slideLeft, slideRight, slideUp, slideDown];
+
+  for (const action of actions) {
+    action(gameBoardWillModified);
+
+    if (gameBoardPrevState.toString() !== gameBoardWillModified.toString()) {
+      return true;
+    }
+  }
+
+  return false;
+};
+
+const gameControl = (e) => {
+  const prevGameBoard = JSON.parse(JSON.stringify(gameBoard));
```

Comment:
You're using `JSON.parse(JSON.stringify(gameBoard))` often, so can move it to a separate function
Or use lodash `deepCopy` since you use lodash anyway